### PR TITLE
Max subarray2

### PR DESCRIPTION
**Pull request by Antonio-Villarreal**
Changed my original push to be a generic function instead of a hard coded example.

- [ ] Add an algorithm?
- [x] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

#### References

[](https://github.com/TheAlgorithms/Java/pull/2917)

### **Checklist:**

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new Java files are placed inside an existing directory.
- [x] All filenames are in all uppercase characters with no spaces or dashes.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by Antonio-Villarreal** [comment]
I fixed the issue you said and made it generic!

**Comment by subhroblkbox** [comment]
Yes and can you move this to previous PR or move that to this PR and add some tests

File: src/main/java/com/thealgorithms/others/MaximumSubarray.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    4. When these three steps are done we continue to maintain the max value.
+
+ */
+import java.util.Scanner;
+
+public class MaximumSubarray{
+
+    public static void main(String[] args) {
+        int x;
+        Scanner scnr = new Scanner(System.in);
```
Comment (subhroblkbox) [inline src/main/java/com/thealgorithms/others/MaximumSubarray.java:26]: in place of taking n from use what you can do is use hasNext from scanner class, along with a List.

**Comment by siriak** [comment]
This algorithm is already implemented here https://github.com/TheAlgorithms/Java/blob/9fb3364cccc5373c32dbdb661f773c5ee753c77a/src/main/java/com/thealgorithms/dynamicprogramming/KadaneAlgorithm.java

**Pull request closed without merging**